app/src/main/java/de/xikolo/controllers/helper/DownloadViewHelper.kt

```diff
@@ -141,13 +152,31 @@ class DownloadViewHelper(
             textFileName.text = downloadAsset.title
         }
 
+        if (description != null) {
+            textDescription.text = description
+            textDescription.visibility = View.VISIBLE
+        } else {
+            textDescription.visibility = View.GONE
+        }
+
         buttonOpenDownload.visibility = View.GONE
 
         size = downloadAsset.size
 
         url = downloadAsset.url
         if (url == null) {
-            view.visibility = View.GONE
+            view.isEnabled = false
+            buttonDownloadStart.isEnabled = false
+
+            if (errorMessage != null) {
+                if (errorMessage != "") {
```

Comment:
Hmm so `null` and `""` are different states for `errorMessage`?

Reply:
Yes, the option to only hide the button when an empty string is supplied makes sense to me.

Reply:
Everything is documented at the very beginning of the `DownloadViewHelper` file.